### PR TITLE
PIM-7927: Fix constraints on attribute code

### PR DESCRIPTION
**Pull request by Doodoune**
**Description (for Contributor and Core Developer)**

Add constraints on attribute code - not allow to create an attribute with code "identifier" (case sensitive) & "attributes".

### CODE REVIEW

File: src/Pim/Bundle/CatalogBundle/Resources/config/validation/attribute.yml [hunk 1]
```diff
@@ -75,7 +75,7 @@ Pim\Bundle\CatalogBundle\Entity\Attribute:
             - Length:
                 max: 255
             - Regex:
-                pattern: /^(?!(id|associationTypes|categories|categoryId|completeness|enabled|family|groups|associations|products|scope|treeId|values|category|parent|(.)*_(products|groups)|entity_type)$)/
+                pattern: /^(?!(id|(?i)\bidentifier\b|associationTypes|categories|categoryId|completeness|enabled|family|groups|associations|products|scope|treeId|values|category|parent|(.)*_(products|groups)|entity_type|attributes)$)/
```
Comment (phaseinducer) [inline src/Pim/Bundle/CatalogBundle/Resources/config/validation/attribute.yml]: I think the `\b` is useless since the pattern is enclosed in ^ and $, no ?

Comment (Doodoune) [reply src/Pim/Bundle/CatalogBundle/Resources/config/validation/attribute.yml]: We doing that for 'family' in 2.3 (https://github.com/akeneo/pim-community-dev/blob/2.3/src/Pim/Bundle/CatalogBundle/Resources/config/validation/attribute.yml)

Comment (Doodoune) [reply src/Pim/Bundle/CatalogBundle/Resources/config/validation/attribute.yml]: I've test, you're right it work without \b, I'll remove it :)

**Comment by Doodoune** [comment]
I close this PR, we will do it on master finally.

**Comment by Doodoune** [comment]
new PR to review on master : https://github.com/akeneo/pim-community-dev/pull/9324

**Pull request closed without merging**